alphabase/smiles/smiles.py

```diff
@@ -0,0 +1,293 @@
+            errors.append(f"Unrecognized N-terminal modification: {n_term_mod}")
+
+        if c_term_mod is not None and c_term_mod not in self.c_term_modifications:
+            errors.append(f"Unrecognized C-terminal modification: {c_term_mod}")
+
+        mol = Chem.MolFromSmiles(aa_smiles)
+        if mol is None:
+            errors.append(f"Invalid amino acid SMILES: {aa_smiles}")
+
+        if errors:
+            raise ValueError("\n".join(errors))
+
+    def modify_amino_acid(
+        self,
+        aa_smiles: str,
+        n_term_mod: Optional[str] = None,
+        c_term_mod: Optional[str] = None,
+    ) -> str:
+        """
+        Modify an amino acid with N-terminal and C-terminal modifications.
+
+        Parameters
+        ----------
+        aa_smiles : str
+            SMILES of an amino acid with or without PTMs, must have the placeholder atoms [Xe] for N-term and [Rn] for C-term.
+        n_term_mod : Optional[str], optional
+            N-terminal modification name. Options are the keys in the n_term_modifications dict.
+            If None, the amino acid is not modified, so using hydrogen atoms.
+        c_term_mod : Optional[str], optional
+            C-terminal modification name. Options are the keys in the c_term_modifications dict.
+            If None, the amino acid is not modified, so using the -OH group.
+
+        Returns
+        -------
+        str
+            SMILES string of the modified amino acid.
+        """
+        self.validate_correct_args(aa_smiles, n_term_mod, c_term_mod)
+        mol = Chem.MolFromSmiles(aa_smiles)
+        n_term_placeholder_mol = Chem.MolFromSmiles(
+            self.N_TERM_PLACEHOLDER, sanitize=False
+        )
+        c_term_placeholder_mol = Chem.MolFromSmiles(
+            self.C_TERM_PLACEHOLDER, sanitize=False
+        )
+
+        mol = self._apply_n_terminal_modification(
+            mol, n_term_placeholder_mol, n_term_mod
+        )
+        mol = self._apply_c_terminal_modification(
+            mol, c_term_placeholder_mol, c_term_mod
+        )
+
+        SanitizeMol(mol)
+        return Chem.MolToSmiles(mol)
+
+    def _apply_n_terminal_modification(
+        self, mol: Chem.Mol, n_term_placeholder_mol: Chem.Mol, n_term_mod: Optional[str]
+    ) -> Chem.Mol:
+        """
+        Apply N-terminal modification to the molecule.
+
+        Parameters
+        ----------
+        mol : Chem.Mol
+            The molecule to modify.
+        n_term_placeholder_mol : Chem.Mol
+            The N-terminal placeholder molecule.
+        n_term_mod : Optional[str]
+            The N-terminal modification to apply.
+
+        Returns
+        -------
+        Chem.Mol
+            The modified molecule.
+        """
+        if n_term_mod:
+            n_mod_smiles = self.n_term_modifications[n_term_mod]
+            n_mod_mol = Chem.MolFromSmiles(n_mod_smiles)
+            mol = ReplaceSubstructs(mol, n_term_placeholder_mol, n_mod_mol)[0]
```

Comment:
I would rename that first `mol` to `original_mol` or smth

Reply:
would not work with the current implementation